cmd/do-agent/config.go

```diff
@@ -141,6 +147,27 @@ func newTimeseriesClient() (*WrappedTSClient, error) {
 	return wrappedTSClient, nil
 }
 
+// getKubernetesClusterUUID retrieves the k8s cluster UUID from the droplet metadata
+func getKubernetesClusterUUID() (string, error) {
+	client := metadata.NewClient(metadata.WithBaseURL(config.metadataURL))
+	userData, err := client.UserData()
+	if err != nil {
+		return "", err
+	}
+	return parseKubernetesClusterUUID(userData), nil
+}
+
+// parseKubernetesClusterUUID parses the user data and returns the value of the kubernetes cluster UUID
+func parseKubernetesClusterUUID(userData string) string {
+	userDataSplit := strings.Split(userData, "\n")
+	for _, line := range userDataSplit {
+		if strings.HasPrefix(line, kubernetesClusterUUIDUserDataPrefix) {
+			return strings.Trim(strings.TrimPrefix(line, kubernetesClusterUUIDUserDataPrefix), "\"")
+		}
+	}
+	return ""
```

Comment:
Should this return some kind of ErrNotFound or something? Will an empty string cause us headaches later if we start getting empty uuids when we don't expect it? :thinking: